frontend/src/scenes/web-analytics/pageReportsLogic.ts

```diff
@@ -77,6 +86,58 @@ export function createUrlPropertyFilter(url: string, stripQueryParams: boolean):
     ]
 }
 
+const INTERVAL_FUNCTIONS: Record<IntervalType, string> = {
+    second: 'toStartOfSecond',
+    minute: 'toStartOfMinute',
+    hour: 'toStartOfHour',
+    day: 'toStartOfDay',
+    week: 'toStartOfWeek',
+    month: 'toStartOfMonth',
+}
+
+const getIntervalFunction = (interval: IntervalType): string => INTERVAL_FUNCTIONS[interval] ?? INTERVAL_FUNCTIONS.day
+
+const createAvgTimeOnPageHogQLQuery = (
+    host: string,
+    pathname: string,
+    filterTestAccounts: boolean,
+    interval: IntervalType,
+    dateRange: { date_from: string | null; date_to: string | null }
+): HogQLQuery => {
+    const intervalFn = getIntervalFunction(interval)
+    return {
+        kind: NodeKind.HogQLQuery,
+        query: `
+SELECT
```

Comment:
I wonder if we should do this differently. I think my first attempt would be to create a `Trends` insight using the `host + prev_pageview_pathname` filter and average the `prev_pageview_duration` to do this without a join, so it also remains a simple Product Analytics Trends insight and the "Open as new Insight" allow people to customize it further than tweaking hogql/sql.

Another reason is that we often get the feedback that it is confusing for users when comparing Product Analytics vs Web Analytics insights, and that they don't match exactly. 

Of course, this query is probably more accurate because of the join with the prev_pageview_id... but it is also slower because of that, so I think we should keep it simple here. WDYT?